tests/unit/test_charm_workload_configuration.py

```diff
@@ -1,189 +1,180 @@
+        self.mock_restart = TestCharmWorkloadConfiguration.patcher_restart.start()
+
+    @staticmethod
+    def teardown() -> None:
+        patch.stopall()
+
+    @pytest.fixture(autouse=True)
+    def create_harness(self, setup, request):
         self.harness = testing.Harness(AUSFOperatorCharm)
-        self.harness.set_model_name(name=self.namespace)
-        self.addCleanup(self.harness.cleanup)
+        self.harness.set_model_name(name=NAMESPACE)
         self.harness.set_leader(is_leader=True)
         self.harness.begin()
+        yield self.harness
+        self.harness.cleanup()
+        request.addfinalizer(self.teardown)
 
-    @patch("charm.check_output")
-    @patch("charm.generate_private_key")
-    def test_given_charm_workload_is_ready_to_configure_and_private_key_is_not_stored_when_update_status_then_private_key_is_generated_and_stored_in_the_container(  # noqa: E501
-        self, patched_generate_private_key, patched_check_output
-    ):
-        test_private_key = b"whatever private key"
-        test_csr = b"whatever csr"
-        self.harness.set_can_connect(container="ausf", val=True)
-        self._create_charm_relations_and_relation_data()
+    @pytest.fixture()
+    def add_storage(self):
         self.harness.add_storage(storage_name="certs", attach=True)
         self.harness.add_storage(storage_name="config", attach=True)
-        patched_check_output.return_value = TEST_POD_IP
-        patched_generate_private_key.return_value = test_private_key
-        root = self.harness.get_filesystem_root("ausf")
-        (root / "support/TLS/ausf.csr").write_text(test_csr.decode())
+
+    @pytest.fixture()
+    def create_charm_relations_and_relation_data(self) -> None:
+        self.harness.add_relation(relation_name="certificates", remote_app="whatever-certs")
+        fiveg_nrf_relation_id = self.harness.add_relation(
+            relation_name="fiveg_nrf", remote_app="whatever-nrf"
+        )
+        self.harness.add_relation_unit(
+            relation_id=fiveg_nrf_relation_id, remote_unit_name="whatever-nrf/0"
+        )
+        self.harness.update_relation_data(
+            relation_id=fiveg_nrf_relation_id,
+            app_or_unit="whatever-nrf",
+            key_values={"url": "https://nrf-example.com:1234"},
+        )
+
+    def test_given_charm_workload_is_ready_to_configure_and_private_key_is_not_stored_when_update_status_then_private_key_is_generated_and_stored_in_the_container(  # noqa: E501
+        self, create_charm_relations_and_relation_data, add_storage
+    ):
+        self.harness.set_can_connect(container=CONTAINER_NAME, val=True)
+        self.mock_check_output.return_value = TEST_POD_IP
+        self.mock_generate_private_key.return_value = TEST_PRIVATE_KEY
+        root = self.harness.get_filesystem_root(CONTAINER_NAME)
+        (root / "support/TLS/ausf.csr").write_text(TEST_CSR.decode())
 
         self.harness.charm.on.update_status.emit()
         self.harness.evaluate_status()
```

Comment:
is evaluate status needed here?
(same for line 119, 139, 161, 182, 211, 236, 275, 296, 321)

Reply:
`evaluate_status` is actually not needed when not checking for a specific status; removing it.